### ISSUE TITLE
Small values leaking through pruning

### ISSUE THREAD

**Issue by mg262** [tensorflow/model-optimization#684]
The documentation for ConstantSparsity and PolynomialDecay suggest that by default, `frequency=100` so that pruning is only applied every 100 steps. This presumably means that the parameters which are pruned to zero will drift away from zero during the other 99/100 steps. So at the end of the pruning process, unless you are careful to have an exact multiple of 100 steps, you well end up with a model that is not perfectly pruned but which has a large number of near-zero values. 

How does one stop this happening? Do you have to tweak `frequency` to be a divisor of the number of steps?  I can't find this in any code samples...

**Comment by liyunlu0618** [comment]
This callback is what you're looking for:
https://github.com/tensorflow/model-optimization/blob/master/tensorflow_model_optimization/python/core/sparsity/keras/pruning_callbacks.py#L64

It makes sure to apply the mask when training finishes.
